src/engine/Minus.cpp

```diff
@@ -95,6 +95,23 @@ size_t Minus::getCostEstimate() {
   return _left->getCostEstimate() + _right->getCostEstimate() + costEstimate;
 }
 
+// _____________________________________________________________________________
+auto Minus::makeUndefRangesChecker(bool left) const {
+  std::variant<ad_utility::Noop, ad_utility::FindSmallerUndefRanges> findUndef;
+  const auto& operation = left ? _left : _right;
+  for (const auto& cols : _matchedColumns) {
+    const auto& [_, info] =
+        operation->getVariableAndInfoByColumnIndex(cols[!left]);
+    // Use expensive operation if one of the columns might contain undef.
+    if (info.mightContainUndef_ !=
+        ColumnIndexAndTypeInfo::UndefStatus::AlwaysDefined) {
+      findUndef.emplace<1>();
+      break;
+    }
+  }
+  return findUndef;
+}
```

Comment:
Also:
There are many cases where we cannot prove statically that there is no UNDEF, but it just isn't there. 
In that case you can just go through all the columns that match to check (you need fully materialized anyway) I am sure I have this implemented in OPTIONAL or MultiColumnJoin if you ne ed inspiration.